test/hotspot/jtreg/testlibrary/ctw/src/sun/hotspot/tools/ctw/ClassPathJimageEntry.java

```
@@ -69,30 +79,40 @@ public void close() {
         }
     }
 
-    private final ImageReader reader;
+    //private final ImageReader reader;
+    private final FileSystem jrtFileSystem;
 
     public ClassPathJimageEntry(Path root) {
         super(root);
         if (!Files.exists(root)) {
             throw new Error(root + " image file not found");
         }
         try {
-            reader = ImageReader.open(root);
+            jrtFileSystem = FileSystemProvider.installedProviders().stream()
+                    .filter(p -> "jrt".equals(p.getScheme()))
+                    .findFirst()
+                    .orElseThrow(() -> new Error("cannot find JRT filesystem for " + root))
+                    .newFileSystem(root, Map.of());
         } catch (IOException e) {
             throw new Error("can not open " + root + " : " + e.getMessage(), e);
         }
     }
 
     @Override
     protected byte[] findByteCode(String name) {
-        String resource = Utils.classNameToFileName(name);
-        for (String m : reader.getModuleNames()) {
-            ImageLocation location = reader.findLocation(m, resource);
-            if (location != null) {
-                return reader.getResource(location);
+        // Relative path to search for inside each /modules/<module> directory.
+        Path resourcePath = jrtFileSystem.getPath(Utils.classNameToFileName(name));
+        Path modulesRoot = jrtFileSystem.getPath("/modules");
+        try (DirectoryStream<Path> modules = Files.newDirectoryStream(modulesRoot)) {
+            for (Path module : modules) {
+                Path p = module.resolve(resourcePath);
+                if (Files.isRegularFile(p)) {
+                    return Files.readAllBytes(p);
+                }
             }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Maybe throw an error for consistency with other methods in this class?